### ISSUE TITLE
puzzle count is differ.

### ISSUE THREAD

**Issue by DronMDF** [yegor256/0pdd#110]
0pdd say 20 puzzles: http://www.0pdd.com/p?name=DronMDF/goost
github have 20 open issues from pdd https://github.com/DronMDF/goost/issues/created_by/0pdd

But pdd say 26 puzzle

```
$ pdd | grep ticket | wc -l
26
```

Why 0pdd not creat issues?

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this issue

**Comment by DronMDF** [comment]
I found problem, but not understand it.
https://github.com/DronMDF/goost/commit/9ab739e74b114943d43c89a9029467a5be0948bd#commitcomment-24152511

**Comment by DronMDF** [comment]
I tried to close todo's, but 0pdd always finds something to break off.
https://github.com/DronMDF/goost/commit/94c1eb49de87ca151c3368c1f4df7c31ff90acc4

**Comment by yegor256** [comment]
@DronMDF it's a bug on our side. For some very strange reason, sometimes 0pdd creates GitHub issues, but doesn't record them in the XML document. Then, later it may create duplicate issues. Issue DronMDF/goost#185 is a good example of such situation. The issue exists in GitHub, but is absent in the XML document. To prevent duplicate issues creation we have a log, which tracks each GitHub operation. `173-f882b25b/submit` is there: http://www.0pdd.com/log?name=DronMDF/goost. That's why no new issues can be created. The line in the log blocks them.

**Comment by yegor256** [comment]
@DronMDF Let me investigate and introduce a fix.

**Comment by yegor256** [comment]
@DronMDF what I see now is that this happens _sometimes_ when a number of puzzles are coming in in a single commit, check these lines in the log:

```
2017-09-09T20:40:15+00:00: 173-f882b25b submitted in issue #185: "Style was failed, cppcheck 1.67 is very old." at .circleci/config.yml; 3-3 /0.22.1 delete
2017-09-09T20:40:14+00:00: 139-df585011 submitted in issue #184: "If EncryptBlock has Block type, we can introduce universal Block representation instead this class" at test2o/kuznyechik/EncryptedBlockTest.cpp; 18-19 /0.22.1 delete
2017-09-09T20:40:13+00:00: 158-6ba0ad65 submitted in issue #183: "LBlock should have BlkRaw interface. Then we can use BlkRaw representation" at test2o/kuznyechik/LBlockTest.cpp; 18-19 /0.22.1 delete
2017-09-09T20:40:12+00:00: 174-b047561a submitted in issue #182: "rename EncryptedBlockTest to BlkEncryptedTest" at test2o/kuznyechik/EncryptedBlockTest.h; 13-13 /0.22.1 delete
```

All four tickets recorded here were created in GitHub, but are absent in XML :(

**Comment by DronMDF** [comment]
By todo id - this is a different commits.
I can analyze this.

**Comment by yegor256** [comment]
@DronMDF IDs are not based on commits, they are just MD5 hash of puzzle texts.

**Comment by DronMDF** [comment]
@yegor256 If i rename file, pdd close old puzzle and open new.
DronMDF/goost#182 puzzle add in c177052d
DronMDF/goost#183 puzzle add in 78b6adfb and changed in 706fd649
DronMDF/goost#184 puzzle add in 23ec6bcc63 and changed twice, last in c177052d
DronMDF/goost#185 puzzle add in 6d53a484

**Comment by yegor256** [comment]
@DronMDF the location of the puzzle shouldn't change anything, since the ID doesn't change. This is how the ID is calculated: https://github.com/yegor256/pdd/blob/0.17.9/lib/pdd/source.rb#L74 Only the "body" (text of the puzzle) is taken into account.

**Comment by DronMDF** [comment]
@yegor256 I change body. By autoreplacment.
This is not a problem.

Renaming files should not lead to the removal and creation of tasks?

**Comment by yegor256** [comment]
@DronMDF no, renaming or moving files should not affect GitHub tickets. If this is now what happens, it's a bug, please report it. But normally, no matter what file the puzzle is located in, it's still one and the same puzzle. Provided, it's text is not changed.

**Comment by DronMDF** [comment]
@yegor256 The problem still remains.
https://github.com/DronMDF/goost/commit/6d13d83d7541f4884e6a25877feed039dd922027#commitcomment-24259412

**Comment by DronMDF** [comment]
@yegor256 0pdd was wailed.

http://www.0pdd.com/p?name=DronMDF/goost

```
 Internal error

undefined local variable or method `name' for #
	/app/0pdd.rb:360:in `storage'
	/app/0pdd.rb:175:in `block in '
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1611:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1611:in `block in compile!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:975:in `block (3 levels) in route!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:994:in `route_eval'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:975:in `block (2 levels) in route!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1015:in `block in process_route'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1013:in `catch'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1013:in `process_route'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:973:in `block in route!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:972:in `each'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:972:in `route!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1085:in `block in dispatch!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1067:in `block in invoke'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1067:in `catch'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1067:in `invoke'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1082:in `dispatch!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:907:in `block in call!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1067:in `block in invoke'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1067:in `catch'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1067:in `invoke'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:907:in `call!'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:895:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-1.5.3/lib/rack/protection/xss_header.rb:18:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-1.5.3/lib/rack/protection/path_traversal.rb:16:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-1.5.3/lib/rack/protection/json_csrf.rb:18:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-protection-1.5.3/lib/rack/protection/frame_options.rb:31:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-1.6.8/lib/rack/logger.rb:15:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-1.6.8/lib/rack/commonlogger.rb:33:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:219:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:212:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-1.6.8/lib/rack/head.rb:13:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-1.6.8/lib/rack/methodoverride.rb:22:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:182:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:2013:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1487:in `block in call'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1787:in `synchronize'
	/app/vendor/bundle/ruby/2.3.0/gems/sinatra-1.4.7/lib/sinatra/base.rb:1487:in `call'
	/app/vendor/bundle/ruby/2.3.0/gems/rack-1.6.8/lib/rack/handler/webrick.rb:88:in `service'
	/app/vendor/ruby-2.3.3/lib/ruby/2.3.0/webrick/httpserver.rb:140:in `service'
	/app/vendor/ruby-2.3.3/lib/ruby/2.3.0/webrick/httpserver.rb:96:in `run'
	/app/vendor/ruby-2.3.3/lib/ruby/2.3.0/webrick/server.rb:296:in `block in start_thread'
```

This is global or project problem?

0pdd not react on new commit.
https://github.com/DronMDF/goost/commit/109052db7707232e7987b1c3609073f149cb775a

**Comment by DronMDF** [comment]
@yegor256 The problem still remains.
https://github.com/DronMDF/goost/commit/dd5248e3740cce30f2da35819a374d56bc5eefd0

Badge still 20/42, not changed.

**Comment by yegor256** [comment]
@rultor release, tag is `0.22.2`

**Comment by yegor256** [comment]
@DronMDF please try now

**Comment by DronMDF** [comment]
@yegor256 Problem still remains.
https://github.com/DronMDF/goost/commit/4124352fbc483dbde24c7dcaae4a7c22e8dc6495

This list is very old. (And badge is very old)
http://www.0pdd.com/p?name=DronMDF/goost

0pdd create #222 (173-f882b25b) where it in puzzle tree?
But this puzzle present in snapshot.

**Comment by yegor256** [comment]
@DronMDF let's try again, I made some major fixes in #116

**Comment by DronMDF** [comment]
Looks okay.
Thanks.

**Comment by 0crat** [comment]
Oops! Job `gh:yegor256/0pdd#110` was not in scope

**Comment by DronMDF** [comment]
@yegor256 Strange... if bug closed, 0pdd should pay +30. or not?
No assigned DEV?

**Comment by yegor256** [comment]
@DronMDF 0crat pays you, not 0pdd :) Yes, +30, if the DEV was assigned.

**Comment by 0pdd** [comment]
@DronMDF all 3 puzzles are solved here: [#111](https://github.com/yegor256/0pdd/issues/111), [#112](https://github.com/yegor256/0pdd/issues/112), [#113](https://github.com/yegor256/0pdd/issues/113).
